openhands/runtime/remote/runtime.py

```diff
@@ -254,14 +248,50 @@ def _parse_runtime_response(self, response: requests.Response):
                 {'X-Session-API-Key': start_response['session_api_key']}
             )
 
-    @retry(
-        stop=stop_after_attempt(60) | stop_if_should_exit(),
-        wait=wait_fixed(2),
-        retry=retry_if_exception_type(RuntimeError),
-        reraise=True,
-    )
     def _wait_until_alive(self):
         logger.info(f'Waiting for runtime to be alive at url: {self.runtime_url}')
+        # send GET request to /runtime/<id>
+        pod_running = False
+        max_not_found_count = 12  # 2 minutes
+        not_found_count = 0
+        while not pod_running:
+            runtime_info_response = send_request_with_retry(
+                self.session,
+                'GET',
+                f'{self.config.sandbox.remote_runtime_api_url}/runtime/{self.runtime_id}',
+                timeout=10,
+            )
+            if runtime_info_response.status_code != 200:
+                raise RuntimeError(
+                    f'Failed to get runtime status: {runtime_info_response.status_code}. Response: {runtime_info_response.text}'
+                )
+            runtime_data = runtime_info_response.json()
+            assert runtime_data['runtime_id'] == self.runtime_id
+            pod_status = runtime_data['pod_status']
+            logger.info(
+                f'Waiting for runtime pod to be active. Current status: {pod_status}'
+            )
+            if pod_status == 'Running':
```

Comment:
I think there's a chance it could be `Running` but not `Ready`--in fact I'm pretty sure that's the case

We should maybe put `Ready` as a new pod_status, and then we wouldn't need to check `/alive` (since Ready waits for the /alive)

Reply:
Yep, this is exactly the reason why i keep the `/alive` call! I think having a `Ready` status (use the same probe we had in the cluster) will be the most ideal.

Reply:
runtime API side now returns `Ready` pod status when readiness probe gives true - this PR should be ready to merge!